### PR TITLE
feat: add body type filter for car listings

### PR DESCRIPTION
**Pull request by Danielsio**
## Summary

- Parse body type (sedan, hatchback, SUV, coupe, wagon, convertible, minivan, pickup) from Yad2 sub-model text using keyword matching
- Store body type in `listing_history` table via new migration (000028)
- Expose `body_type` field in all listing API responses
- Add client-side filter chips with Hebrew labels in the listings filter bar

## Review

✅ 7/7 agents passed (correctness, testing, maintainability, api-contract, data-migrations, adversarial, reliability)

No P0/P1 issues found. P2 findings are design choices (no backfill for existing rows — populated on next scrape) or pre-existing (response builder duplication).

## Implementation

| Layer | Changes |
|-------|---------|
| **Parser** | New `internal/bodytype/` package with `Parse()` function — keyword matching with word boundary support for short tokens |
| **Model** | `BodyType string` field on `RawListing` |
| **Database** | Migration 000028: `ALTER TABLE listing_history ADD COLUMN body_type TEXT NOT NULL DEFAULT ''` |
| **Storage** | `body_type` wired through all SQL queries (upsert, scan, 5 SELECT queries, admin) |
| **API** | `body_type` field added to `listingResponse` JSON (omitempty) |
| **Frontend** | Filter chips with Hebrew labels, facet derivation, client-side filtering |

## Test plan

- [x] `internal/bodytype/bodytype_test.go` — 41 test cases covering all body types, Hebrew/English keywords, word boundaries, edge cases
- [x] `internal/fetcher/yad2/parser_test.go` — 2 new tests verifying body type extraction from feed data
- [x] `web/src/hooks/useListingFilters.test.ts` — 3 new tests for filtering + facet derivation
- [x] `web/src/components/ListingsFilterBar.test.tsx` — 2 new tests for chip rendering + toggling
- [x] Go build passes
- [x] All frontend tests pass (25/25)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Vehicle listings now include body type information throughout the app.
  * Added body-type filtering in search and listings browsing, with quick-select chips in the filter bar.
  * Body-type labels are now shown in Hebrew in the UI.

* **Bug Fixes**
  * Improved listing data display so body type appears consistently in listing details, search results, saved items, and admin views.
  * Added support for extracting body type from listing text when available.

### CODE REVIEW

File: internal/bodytype/bodytype.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	Hatchback   = "hatchback"
+	SUV         = "suv"
+	Coupe       = "coupe"
+	Wagon       = "wagon"
+	Convertible = "convertible"
+	Minivan     = "minivan"
+	Pickup      = "pickup"
+)
+
+var All = []string{Sedan, Hatchback, SUV, Coupe, Wagon, Convertible, Minivan, Pickup}
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply internal/bodytype/bodytype.go]: Fixed in 9e5fd1a — replaced mutable `All` slice with `All()` function returning a copy, and replaced `init()` with a static literal for the matchers slice.
